src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/DiagnosticUtilities.cs

```diff
@@ -43,5 +43,41 @@ internal static string GetGenericParameterDeclaringMemberDisplayName(GenericPara
             else
                 return ((TypeDesc)parent).GetDisplayName();
         }
+
+        internal static string GetRequiresUnreferencedCodeAttributeMessage(MethodDesc method)
+        {
+            var ecmaMethod = method.GetTypicalMethodDefinition() as EcmaMethod;
+            if (ecmaMethod == null)
+                return null;
+
+            var decoded = ecmaMethod.GetDecodedCustomAttribute("System.Diagnostics.CodeAnalysis", "RequiresUnreferencedCodeAttribute");
+            if (decoded == null)
+                return null;
+
+            var decodedValue = decoded.Value;
+
+            if (decodedValue.FixedArguments.Length != 0)
+                return (string)decodedValue.FixedArguments[decodedValue.FixedArguments.Length - 1].Value;
```

Comment:
Is there a prior art for using the last argument instead of the first argument as the message?

It does not matter today since the attribute constructor is not overloaded. If we were to overload the attribute constructor, the message is likely going to stay as the first argument and the additional arguments will go after it.

Reply:
Ah. That's copy-paste from how we parse RuntimeImport. That's definitely a pattern that wouldn't pass the API review board.